File: macOS/ViewModels/WelcomeViewModel.swift
import Foundation
import Combine

class WelcomeViewModel: ObservableObject {
    @Published var greeting = "Hello!"
    var onNavigationRequested: (() -> Void)?

    init() {
        startWelcomeSequence()
    }

    private func startWelcomeSequence() {
        // Reduced hold time for a faster feel
        DispatchQueue.main.asyncAfter(deadline: .now() + 2.0) {
            self.onNavigationRequested?()
        }
    }
}

File: Windows/ViewModels/WelcomeViewModel.cs
using System;
using System.ComponentModel;
using System.Runtime.CompilerServices;
using System.Threading.Tasks;
using System.Windows.Input;

namespace StartupDashboard.ViewModels
{
    public class WelcomeViewModel : INotifyPropertyChanged
    {
        private string _greeting = "Hello Noah";
        public string Greeting
        {
            get => _greeting;
            set { _greeting = value; OnPropertyChanged(); }
        }

        public event Action? NavigationRequested;

        public WelcomeViewModel()
        {
            StartWelcomeSequence();
        }

        private async void StartWelcomeSequence()
        {
            // Reduced hold time for a faster, snappier feel
            await Task.Delay(1500);
            NavigationRequested?.Invoke();
        }

        public event PropertyChangedEventHandler? PropertyChanged;
        protected void OnPropertyChanged([CallerMemberName] string? name = null)
        {
            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(name));
        }
    }
}
